fuel-core/src/schema/block.rs

```diff
@@ -79,6 +84,19 @@ impl Block {
         &self.header
     }
 
+    async fn signature(&self, ctx: &Context<'_>) -> Option<Signature> {
```

Comment:
I think we should have something more generic than signature here, as PoS will have multiple signatures etc. Can we reuse the fuel block consensus type instead?

Reply:
Of course, we can=) I did it as a signature because it is easy to use. The question do we want to bring the complexity now or later=)

Reply:
After a private discussion, we can add complexity now and add an additional `block_producer` method to the `Block`.